plugins/pipelines/update.js

```diff
@@ -136,6 +141,32 @@ module.exports = () => ({
             // update pipeline
             const updatedPipeline = await oldPipeline.update();
 
+            // check if pipeline has deploy key annotation then create secrets
+            const deployKeyAnnotation = oldPipeline.annotations && oldPipeline.annotations[ANNOTATION_USE_DEPLOY_KEY]
+
+            if (deployKeyAnnotation) {
+                const deploySecret = await secretFactory.get({
+                    pipelineId: oldPipeline.id,
+                    name: deployKeySecret,
+                })
+
+                if (!deploySecret) {
+                    const privateDeployKey = await pipelineFactory.scm.addDeployKey({
+                        scmContext : oldPipeline.scmContext,
+                        checkoutUrl: formattedCheckoutUrl,
+                        token
+                    });
+                    const privateDeployKeyB64 = Buffer.from(privateDeployKey).toString('base64');
+
+                    await secretFactory.create({
```

Comment:
should this have a try/catch with error logs?

Reply:
seems this doesn't throw an error, we are anyway checking if the secret already exists before creating, also API call  should be logged already